feat: add C implementation for `stats/base/dists/arcsine/median`

pranav-1720 opened the pull request:
Progresses [#3403](https://github.com/stdlib-js/stdlib/issues/3403).

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds C implementation for @stdlib/stats/base/dists/arcsine/median

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses [#3355 ](https://github.com/stdlib-js/stdlib/issues/3355)

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

pranav-1720 committed:
feat: add C implementation for @stdlib/stats/base/dists/arcsine/median

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/README.md:
```diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
+
+</section>
+
+<!-- /.c -->
+
+
+
 <!-- Section to include cited references. If references are included, add a horizontal rule *before* the section. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
```
```suggestion
<!-- /.c -->

<!-- Section to include cited references. If references are included, add a horizontal rule *before* the section. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/README.md:
````diff
@@ -133,6 +133,96 @@ for ( i = 0; i < 10; i++ ) {
+double stdlib_base_dists_arcsine_median( const double a, const double b );
+```
+
+</section>
+<!-- /.usage -->
+<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+<section class="notes">
+</section>
+<!-- /.notes -->
+<!-- C API usage examples. -->
+<section class="examples">
````
```suggestion
</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,62 @@
+
+// MAIN //
+
+bench( pkg, opts, function benchmark( b ) {
```
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/include/stdlib/stats/base/dists/arcsine/median.h:
```diff
@@ -0,0 +1,39 @@
+}
+#endif
+
+#endif // !STDLIB_STATS_BASE_DISTS_ARCSINE_MEDIAN_H
```
```suggestion
#endif // !STDLIB_STATS_BASE_DISTS_ARCSINE_MEDIAN_H

```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/manifest.json:
```diff
@@ -0,0 +1,72 @@
+        ]
+      }
+    ]
+  }
```
```suggestion
  }
  
```

pranav-1720 committed:
Update lib/node_modules/@stdlib/stats/base/dists/arcsine/median/README.md

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: pranav-1720 <123018993+pranav-1720@users.noreply.github.com>

pranav-1720 committed:
Update lib/node_modules/@stdlib/stats/base/dists/arcsine/median/README.md

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: pranav-1720 <123018993+pranav-1720@users.noreply.github.com>

pranav-1720 committed:
Update lib/node_modules/@stdlib/stats/base/dists/arcsine/median/benchmark/benchmark.native.js

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: pranav-1720 <123018993+pranav-1720@users.noreply.github.com>

pranav-1720 committed:
Update lib/node_modules/@stdlib/stats/base/dists/arcsine/median/include/stdlib/stats/base/dists/arcsine/median.h

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: pranav-1720 <123018993+pranav-1720@users.noreply.github.com>

pranav-1720 committed:
Update lib/node_modules/@stdlib/stats/base/dists/arcsine/median/manifest.json

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: pranav-1720 <123018993+pranav-1720@users.noreply.github.com>

Planeshifter committed:
chore: add missing metadata and fix formatting

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/src/main.c:
```diff
@@ -0,0 +1,36 @@
+*
+* @param a    minimum support
+* @param b    maximum support
+* @return     median
+* @example
+* double y = stdlib_base_arcsine_median( 0.0, 1.0 );
+* // returns 0.5
+*/
```
```suggestion
* @return     median
*
* @example
* double y = stdlib_base_dists_arcsine_median( 0.0, 1.0 );
* // returns 0.5
*/
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/include/stdlib/stats/base/dists/arcsine/median.h:
```diff
@@ -0,0 +1,39 @@
+/**
+* Returns the median of an arcsine distribution.
+*/
+double stdlib_base_dists_arcsine_median(const double a, const double b );
```
```suggestion
double stdlib_base_dists_arcsine_median( const double a, const double b );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/arcsine/median/lib/native.js:
```diff
@@ -0,0 +1,66 @@
+
+/**
+* Returns the median of an arcsine distribution.
+*
+* @param {number} a - minimum support
```
```suggestion
*
* @private
* @param {number} a - minimum support
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thanks @pranav-1720! Made a few edits but will merge once all CI checks are confirmed as green.
